bundles/automation/org.eclipse.smarthome.automation.module.test/src/test/groovy/org/eclipse/smarthome/automation/module/RuntimeRuleTest.groovy

```diff
@@ -14,172 +14,172 @@ import static org.hamcrest.CoreMatchers.*
-		]
-
-		def rule = automationFactory.createRule(triggers, conditions, actions, null, null)
-		// I would expect the factory to create the UID of the rule and the name to be in the list of parameters.
-		rule.UID="myRule21"
-		rule.name="RuleByJAVA_API"
-
-		logger.info("Rule created: "+rule.getUID())
-
-		def ruleRegistry = getService(RuleRegistry)
-		ruleRegistry.add(rule)
-		ruleRegistry.setEnabled(rule.UID, true)
-
-		//TEST RULE
-
-		def EventPublisher eventPublisher = getService(EventPublisher)
-		def ItemRegistry itemRegistry = getService(ItemRegistry)
-		SwitchItem myMotionItem = itemRegistry.getItem("myMotionItem2")
-		Command commandObj = TypeParser.parseCommand(myMotionItem.getAcceptedCommandTypes(), "ON")
-		eventPublisher.post(ItemEventFactory.createCommandEvent("myPresenceItem2", commandObj))
-		//		eventPublisher.post(ItemEventFactory.createStateEvent("myPresenceItem2", OnOffType.ON))
-
-		Event event = null
-
-		def eventHandler = [
-			receive: {  Event e ->
-				logger.info("Event: " + e.topic)
-				if (e.topic.contains("myLampItem2")){
-					event=e
-				}
-			},
-
-			getSubscribedEventTypes: {
-				Sets.newHashSet(ItemUpdatedEvent.TYPE, ItemStateEvent.TYPE)
-			},
-
-			getEventFilter:{ null }
-
-		] as EventSubscriber
-
-		registerService(eventHandler)
-		commandObj = TypeParser.parseCommand(itemRegistry.getItem("myMotionItem2").getAcceptedCommandTypes(),"ON")
-		eventPublisher.post(ItemEventFactory.createCommandEvent("myMotionItem2", commandObj))
-		//		eventPublisher.post(ItemEventFactory.createStateEvent("myMotionItem2", OnOffType.ON))
-		waitForAssert ({ assertThat event, is(notNullValue())} , 3000, 100)
-		assertThat event.topic, is(equalTo("smarthome/items/myLampItem2/state"))
-		assertThat (((ItemStateEvent)event).itemState, is(OnOffType.ON))
-		def myLampItem2 = itemRegistry.getItem("myLampItem2")
-		assertThat myLampItem2, is(notNullValue())
-		logger.info("myLampItem2 State: " + myLampItem2.state)
-		assertThat myLampItem2.state, is(OnOffType.ON)
-	}
+    final Logger logger = LoggerFactory.getLogger(RuntimeRuleTest.class)
+    VolatileStorageService volatileStorageService = new VolatileStorageService()
+
+    @Before
+    void before() {
+        def itemProvider = [
+            getAll: {
+                [
+                    new SwitchItem("myMotionItem"),
+                    new SwitchItem("myPresenceItem"),
+                    new SwitchItem("myLampItem"),
+                    new SwitchItem("myMotionItem2"),
+                    new SwitchItem("myPresenceItem2"),
+                    new SwitchItem("myLampItem2")
+                ]
+            },
+            addProviderChangeListener: {},
+            removeProviderChangeListener: {},
+            allItemsChanged: {}] as ItemProvider
+        registerService(itemProvider)
+        registerService(volatileStorageService)
+
+        def autoupdateConfig = [
+            autoUpdate: { String itemName -> return true }
```

Comment:
that´s interesting. we should add this to the base class. Like a method "setAutoUpdateEnabled(enable)"